iotdb-core/datanode/src/main/java/org/apache/iotdb/db/queryengine/plan/relational/planner/node/CreateTableDeviceNode.java

```diff
@@ -0,0 +1,236 @@
+    PlanNodeType.CREATE_TABLE_DEVICE.serialize(byteBuffer);
+    ReadWriteIOUtils.write(devicePathList.size(), byteBuffer);
+    for (PartialPath deviceId : devicePathList) {
+      deviceId.serialize(byteBuffer);
+    }
+    ReadWriteIOUtils.write(attributeNameList.size(), byteBuffer);
+    for (String attributeName : attributeNameList) {
+      ReadWriteIOUtils.write(attributeName, byteBuffer);
+    }
+    ReadWriteIOUtils.write(attributeValueList.size(), byteBuffer);
+    for (List<String> deviceValueList : attributeValueList) {
+      ReadWriteIOUtils.write(deviceValueList.size(), byteBuffer);
+      for (String value : deviceValueList) {
+        ReadWriteIOUtils.write(value, byteBuffer);
+      }
+    }
+  }
+
+  @Override
+  protected void serializeAttributes(DataOutputStream stream) throws IOException {
+    PlanNodeType.CREATE_TABLE_DEVICE.serialize(stream);
+    ReadWriteIOUtils.write(devicePathList.size(), stream);
+    for (PartialPath deviceId : devicePathList) {
+      deviceId.serialize(stream);
+    }
+    ReadWriteIOUtils.write(attributeNameList.size(), stream);
+    for (String attributeName : attributeNameList) {
+      ReadWriteIOUtils.write(attributeName, stream);
+    }
+    for (List<String> deviceValueList : attributeValueList) {
+      for (String value : deviceValueList) {
+        ReadWriteIOUtils.write(value, stream);
+      }
+    }
+  }
+
+  public static CreateTableDeviceNode deserialize(ByteBuffer buffer) {
+    int deviceNum = ReadWriteIOUtils.readInt(buffer);
+    List<PartialPath> devicePathList = new ArrayList<>(deviceNum);
+    for (int i = 0; i < deviceNum; i++) {
+      devicePathList.add((PartialPath) PathDeserializeUtil.deserialize(buffer));
+    }
+    int attributeNameNum = ReadWriteIOUtils.readInt(buffer);
+    List<String> attributeNameList = new ArrayList<>(attributeNameNum);
+    for (int i = 0; i < attributeNameNum; i++) {
+      attributeNameList.add(ReadWriteIOUtils.readString(buffer));
+    }
+    List<List<String>> attributeValueList = new ArrayList<>(deviceNum);
+    for (int i = 0; i < deviceNum; i++) {
+      List<String> deviceValueList = new ArrayList<>(attributeNameNum);
+      for (int j = 0; j < attributeNameNum; j++) {
+        deviceValueList.add(ReadWriteIOUtils.readString(buffer));
+      }
+      attributeValueList.add(deviceValueList);
+    }
+    PlanNodeId planNodeId = PlanNodeId.deserialize(buffer);
+    return new CreateTableDeviceNode(
+        planNodeId, devicePathList, attributeNameList, attributeValueList);
+  }
+
+  @Override
+  public List<WritePlanNode> splitByPartition(IAnalysis analysis) {
+    Map<TRegionReplicaSet, List<Integer>> splitMap = new HashMap<>();
+    for (int i = 0; i < devicePathList.size(); i++) {
+      TRegionReplicaSet regionReplicaSet =
+          analysis
+              .getSchemaPartitionInfo()
+              .getSchemaRegionReplicaSet(new StringArrayDeviceID(devicePathList.get(i).getNodes()));
```

Comment:
pass String database and use `IDeviceID.Factory.create(String[])` to create IDeviceId.

Reply:
add new method in getSchemaRegionReplicaSet(String database, IDeviceId)

Reply:
Fixed implementation.